confmap/resolver.go

```diff
@@ -242,44 +239,25 @@ func (mr *Resolver) expandValueRecursively(ctx context.Context, value any) (any,
 func (mr *Resolver) expandValue(ctx context.Context, value any) (any, bool, error) {
 	switch v := value.(type) {
 	case string:
-		// If no embedded "uris" no need to expand. embeddedURI regexp matches uriRegexp as well.
-		if !embeddedURI.MatchString(v) {
+		// No URIs to expand.
+		if !strings.Contains(v, "${") || !strings.Contains(v, "}") {
 			return value, false, nil
 		}
 
+		// Don't expand. Too many closing brackets.
+		if strings.Count(v, "}") > 100 {
```

Comment:
Is this to protect from too deep recursion?

Reply:
I also thought of throwing an error here, but this would be a breaking change, as previously using > 100 `}` (or > 100 embedded URIs) wouldn't throw an error. I've documented the 100 URI limit [here](https://github.com/open-telemetry/opentelemetry-collector/blob/e283dced6215def786b51fd91261e9e7a16914bf/confmap/README.md#configuration-resolving) (in limitation).